---  quodlibet/__init__.py.old	2014-08-16 20:07:31.000000000 +0200
---  quodlibet/__init__.py	2014-08-16 20:07:31.000000000 +0200
@@ -584,7 +584,13 @@
 
         print_d("Quit GTK: done.")
 
+    # START MAC OS X STUFF
     window.connect('destroy', quit_gtk)
+    #window.macapp.connect('NSApplicationBlockTermination', quit_gtk)
+    #window.macapp.connect('NSApplicationWillTerminate', quit_gtk)
+    window.macapp.ready()
+    # END MAC OS X STUFF
+    
     window.show_maybe()
 
     Gtk.main()
--- quodlibet/qltk/quodlibetwindow.py.old	2014-08-16 20:28:32.000000000 +0200
--- quodlibet/qltk/quodlibetwindow.py	2014-08-16 20:28:32.000000000 +0200
@@ -382,6 +382,23 @@
         accel_group = ui.get_accel_group()
         self.add_accel_group(accel_group)
 
+        # BEGIN MAC OS X STUFF
+        menu = ui.get_widget("/Menu")
+        from gi.repository import GtkosxApplication as gtkosx
+        self.macapp = gtkosx.Application()
+
+        menu.hide()
+        self.macapp.set_menu_bar(menu)
+        # Reparent some items to the "Application" menu
+        item = ui.get_widget('/Menu/Help/About')
+        self.macapp.insert_app_menu_item(item, 0)
+        self.macapp.insert_app_menu_item(Gtk.SeparatorMenuItem(), 1)
+        item = ui.get_widget('/Menu/Music/Preferences')
+        self.macapp.insert_app_menu_item(item, 2)
+        quit_item = ui.get_widget('/Menu/Music/Quit')
+        quit_item.hide()
+        # END MAC OS X STUFF
+
         # dbus app menu
         AppMenu(self, ui.get_action_groups()[0])
 
@@ -391,7 +409,7 @@
         def accel_save_cb(*args):
             Gtk.AccelMap.save(accel_fn)
         accel_group.connect_object('accel-changed', accel_save_cb, None)
-        main_box.pack_start(ui.get_widget("/Menu"), False, True, 0)
+        #main_box.pack_start(ui.get_widget("/Menu"), False, True, 0)
 
         # get the playlist up before other stuff
         self.songlist = MainSongList(library, player)
@@ -680,13 +698,13 @@
         actions = [
             ('Music', None, _("_Music")),
             ('AddFolders', Gtk.STOCK_ADD, _('_Add a Folder...'),
-             "<control>O", None, self.open_chooser),
+             "<meta>O", None, self.open_chooser),
             ('AddFiles', Gtk.STOCK_ADD, _('_Add a File...'),
              None, None, self.open_chooser),
             ('AddLocation', Gtk.STOCK_ADD, _('_Add a Location...'),
              None, None, self.open_location),
             ('BrowseLibrary', Gtk.STOCK_FIND, _('Open _Browser'), ""),
-            ("Preferences", Gtk.STOCK_PREFERENCES, None, None, None,
+            ("Preferences", Gtk.STOCK_PREFERENCES, None, '<meta>comma', None,
              self.__preferences),
             ("Plugins", Gtk.STOCK_EXECUTE, _("_Plugins"), None, None,
              self.__plugins),
@@ -709,7 +727,7 @@
              self.__current_song_info),
 
             ("Jump", Gtk.STOCK_JUMP_TO, _("_Jump to Playing Song"),
-             "<control>J", None, self.__jump_to_current),
+             "<meta>J", None, self.__jump_to_current),
 
             ("View", None, _("_View")),
             ("Help", None, _("_Help")),
@@ -718,13 +736,13 @@
             ]
 
         actions.append(("Previous", Gtk.STOCK_MEDIA_PREVIOUS, None,
-                        "<control>comma", None, self.__previous_song))
+                        "<ctrl>comma", None, self.__previous_song))
 
         actions.append(("PlayPause", Gtk.STOCK_MEDIA_PLAY, None,
-                        "<control>space", None, self.__play_pause))
+                        "<ctrl>space", None, self.__play_pause))
 
         actions.append(("Next", Gtk.STOCK_MEDIA_NEXT, None,
-                        "<control>period", None, self.__next_song))
+                        "<ctrl>period", None, self.__next_song))
 
         ag.add_actions(actions)
 
@@ -739,12 +757,12 @@
             "AddBookmark", _("Add Bookmark"), None, Gtk.STOCK_ADD)
         act.connect_object('activate', self.__add_bookmark,
                            library.librarian, player)
-        ag.add_action_with_accel(act, "<ctrl>D")
+        ag.add_action_with_accel(act, "<meta>D")
 
         act = Gtk.Action.new("EditBookmarks", _(u"Edit Bookmarks…"), None, "")
         act.connect_object('activate', self.__edit_bookmarks,
                            library.librarian, player)
-        ag.add_action_with_accel(act, "<ctrl>B")
+        ag.add_action_with_accel(act, "<meta>B")
 
         act = Gtk.Action.new("About", None, None, Gtk.STOCK_ABOUT)
         act.connect_object('activate', self.__show_about, player)
@@ -753,7 +771,7 @@
         act = Gtk.Action.new(
             "OnlineHelp", _("Online Help"), None, Gtk.STOCK_HELP)
         act.connect_object('activate', util.website, const.ONLINE_HELP)
-        ag.add_action_with_accel(act, "F1")
+        ag.add_action_with_accel(act, "<meta>question")
 
         act = Gtk.Action.new("SearchHelp", _("Search Help"), None, "")
         act.connect_object('activate', util.website, const.SEARCH_HELP)
@@ -781,7 +799,7 @@
             act = Gtk.Action.new("Random%s" % util.capitalize(tag_), label,
                                  None, Gtk.STOCK_DIALOG_QUESTION)
             act.connect('activate', self.__random, tag_)
-            ag.add_action_with_accel(act, "<control>" + accel)
+            ag.add_action_with_accel(act, "<meta>" + accel)
 
         ag.add_toggle_actions([
             ("SongList", None, _("Song _List"), None, None,
